pandas/tests/indexes/test_base.py

```diff
@@ -1557,6 +1557,41 @@ def test_string_index_repr(self):
 
                 self.assertEqual(coerce(idx), expected)
 
+    def test_map(self):
```

Comment:
put all of this in ``pandas/tests/index/common.py/Base`` it will then get tested on each index type. see the structure of other methods. If you need to override it then put it specifically in a ``pandas/test/index/test_*`` file.